python/pyspark/pandas/frame.py

```diff
@@ -2563,12 +2563,18 @@ def apply_func(pdf: pd.DataFrame) -> pd.DataFrame:
 
             index_spark_columns = None
             index_names: Optional[List[Optional[Tuple[Any, ...]]]] = None
-            index_fields = None
+
             if should_retain_index:
-                index_spark_columns = [scol_for(sdf, index_field.struct_field.name)]
-                index_fields = [index_field]
-                if index_field.struct_field.name != SPARK_DEFAULT_INDEX_NAME:
-                    index_names = [(index_field.struct_field.name,)]
+                index_spark_columns = [
+                    scol_for(sdf, index_field.struct_field.name) for index_field in index_fields
+                ]
+                if all(
+                    [
+                        index_field.struct_field.name != SPARK_DEFAULT_INDEX_NAME
```

Comment:
Maybe we should use `SPARK_INDEX_NAME_PATTERN` and if the pattern matches.

Reply:
Thanks, updated, please take another look when you have time!